### PR TITLE
Add cluster-checks documentation

### PR DESCRIPTION
**Pull request by xvello**
### What does this PR do?

Document the upcoming Cluster Checks feature that will ship with the 1.2.0 cluster-agent.

### Preview link

https://docs-staging.datadoghq.com/xvello/cchecks/agent/autodiscovery/clusterchecks/
https://docs-staging.datadoghq.com/xvello/cchecks/agent/kubernetes/cluster/#cluster-checks-autodiscovery

### CODE REVIEW

File: content/agent/autodiscovery/clusterchecks.md [hunk 1]
```diff
@@ -0,0 +1,257 @@
...
+---
+
+## How it Works
+
+The Datadog Agent is able to auto-discover containers and create check configurations based on them via [the Autodiscovery mechanism][1]. The Cluster Checks feature extends this mechanism to monitor non-containerized workloads, including:
+
+- out-of-cluster datastores and endpoints (eg. RDS or CloudSQL)
+- load-balanced cluster services (eg. Kubernetes services)
+
+To ensure only one instance of each check runs, [the Cluster Agent][2] holds the configurations and dynamically dispatches them to node-based Agents. The Agents will connect to the Cluster Agent every 10 seconds and retrieve configurations to run. If an Agent stops reporting, the Cluster Agent will remove it from it's active pool and dispatch the configurations to other Agents. This ensures one instance always runs even as you add and remove nodes from your cluster.
```
[the remark this answers is not in the document]
Comment (hkaj) [reply content/agent/autodiscovery/clusterchecks.md]: `taht` --> `that`
`one (and only one) instance always even as nodes are added` --> `always **runs**, even`?

File: content/agent/autodiscovery/clusterchecks.md [hunk 2]
```diff
@@ -0,0 +1,257 @@
...
+
+Metrics, events and service checks collected by Cluster Checks will be submitted without a hostname, as it is not relevant. A `cluster_name` tag is added, to allow you to scope and slice your data.
```
[the remark this answers is not in the document]
Comment (hkaj) [reply content/agent/autodiscovery/clusterchecks.md]: `service checks`, or `Service Checks` as this is a specific concept.

File: content/agent/autodiscovery/clusterchecks.md [hunk 3]
```diff
@@ -0,0 +1,265 @@
...
+  ad.datadoghq.com/service.instances: '[<INSTANCE_CONFIG>]'
+```
+
+The `%%host%%` [template variable][9] is supported and will be replaced by the service's IP. The `kube_namespace` and `kube_service` tags will be added to the instance.
+
+#### Example: HTTP check on an nginx-backed service
+
+The following Service definition will expose the Pods from the `my-nginx` deployment and run an [HTTP check][10] to measure the latency of the load-balanced service:
+
+```
```
Comment (hkaj) [inline content/agent/autodiscovery/clusterchecks.md]: dunno if we color stuff in our docs, but in case we do (or plan on doing it in the future), we could add `yaml` after the 3 backquotes here.

File: content/agent/autodiscovery/clusterchecks.md [hunk 4]
```diff
@@ -0,0 +1,265 @@
...
+      ]
+spec:
+  ports:
+  - port: 80
+    protocol: TCP
+  selector:
+    run: my-nginx
+```
+
+In addition, each pod should be monitored with the [NGINX check][11], as it allows to observe both each worker and the aggregated service.
```
Comment (hkaj) [inline content/agent/autodiscovery/clusterchecks.md]: `both each worker and the aggregated service.` --> `each worker as well as the aggregated service`

File: content/agent/autodiscovery/clusterchecks.md [hunk 5]
```diff
@@ -0,0 +1,257 @@
...
+Init Config:
+{}
+Auto-discovery IDs:
+* kube_service://751adfe4-1280-11e9-a26b-42010a9c00c8
+===
+```
+
+### Dispatching logic in the Cluster Agent
+
+The `clusterchecks` command allows to inspect the state of the dispatching logic, and track on what node a configuration will be running.
```
[the remark this answers is not in the document]
Comment (hkaj) [reply content/agent/autodiscovery/clusterchecks.md]: I think this is about the mapping between cluster checks to run, and where (i.e. on which node agent they're currently running). But yeah we should maybe rephrase that.

File: content/agent/kubernetes/cluster.md [hunk 6]
```diff
@@ -248,6 +260,20 @@ To enable the Custom Metrics Server:
 
 Refer to [the dedicated Custom metrics server guide][1] to configure the Custom Metrics Server and get more details about this feature.
 
+#### Cluster Checks Autodiscovery
+
+Starting with version 1.2.0, the Datadog Cluster Agent can extend the Autodiscovery mechanism for non-containerized cluster resources. To enable this, make the following changes to the Cluster Agent deployment:
+
+1. Set `DD_CLUSTER_CHECKS_ENABLED` to `true`.
+1. Pass your cluster name as `DD_CLUSTER_NAME`. It will be injected as a `cluster_name` instance tag to all configurations, to help you scope your metrics. The tag name can be customized with the `DD_CLUSTER_CHECKS_CLUSTER_TAG_NAME` option.
```
Comment (hkaj) [inline content/agent/kubernetes/cluster.md]: all of these are `1.`s

File: content/agent/autodiscovery/clusterchecks.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (xvello) [reply content/agent/autodiscovery/clusterchecks.md]: Reworded

**Comment by xvello** [comment]
Merge delayed until Cluster Agent 1.2.0 is released

**Comment by xvello** [comment]
All good after rebase, let's merge this 🌮
